tests/s25Main/integration/testGamePlayer.cpp

```diff
@@ -65,7 +65,7 @@ BOOST_FIXTURE_TEST_CASE(ProductivityStats, WorldFixtureEmpty1P)
     MapPoint curPos(0, 0);
     for(const auto bldType : helpers::EnumRange<BuildingType>{})
     {
-        if(!BuildingProperties::IsValid(bldType))
+        if(!BuildingProperties::IsValid(bldType) || bldType == BuildingType::HarborBuilding)
```

Comment:
Not sure what exactly is the failing test 'ProductivityStats' testing.
However it probably fails, because last created building Harbour is recalculating border and destroying buildings. 

Just skipping the creation of harbour will fix the test

Reply:
That would make sense, yes. Maybe we need a comment why this is OK though:
```suggestion
        if(!BuildingProperties::IsValid(bldType))
            continue;
        if(bldType == BuildingType::HarborBuilding)
        {
            // Creating a harbor would recalculate the territory and destroy some buildings already added in the calculation, but we don't have productivity for it so no need to test
            BOOST_TEST_REQUIRE(!helpers::contains(iwBuildingProductivities::icons, bldType))
            continue
        }
```

However I don't feel so good about excluding (real) buildings here as we might miss bugs. Maybe a better solution would be to convert `helpers::EnumRange<BuildingType>` into a vector and sort such that all "usual" buildings come last, i.e. `BuildingProperties::IsUsual`

This way territory changes only happen at the start